file/file.go

```diff
@@ -0,0 +1,70 @@
+// Package file - Handle file operations
+package file
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+)
+
+const filename = "tasks.json"
+
+// File - file generic struct. T generic.
+type File[T any] struct{}
+
+// New - instantiates a File struct.
+func New[T any]() *File[T] {
+	return &File[T]{}
+}
+
+// Write - writes tasks into a JSON file
+func (f *File[T]) Write(data *[]T) error {
+	asJSON, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(filename, asJSON, 0644)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Read - reads tasks from json file
+func (f *File[T]) Read() ([]T, error) {
+	jsonFile, err := os.Open(filename)
+
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			jsonFile, err = os.Create(filename)
+
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			return nil, err
```

Comment:
If in some situation we needed to open file and then create it if it's not exists, we can do it with less nesting, which is simpler.

```
f, err := os.Open(filename) // no empty line after that
if os.IsNotExist(err) { // there is even shorter version then error.Is(...)
    f, err = os.Create(filename)
    if err != nil {
        return fmt.Errorf("create file: %w", err)
    }
}
if err != nil {
    return fmt.Errorf("open file: %w", err)
}
```